pkg/datapath/iptables/iptables.go

```diff
@@ -202,171 +242,173 @@ func InstallRules(ifName string) error {
-	if err := runProg("iptables", []string{
-		"-t", "filter",
-		"-A", ciliumOutputChain,
-		"-m", "mark", "!", "--mark", matchFromIPSecDecrypt, // Don't match ipsec traffic
-		"-m", "mark", "!", "--mark", matchFromIPSecEncrypt, // Don't match ipsec traffic
-		"-m", "mark", "!", "--mark", matchFromProxy, // Don't match proxy traffic
-		"-m", "comment", "--comment", "cilium: host->any mark as from host",
-		"-j", "MARK", "--set-xmark", markAsFromHost}, false); err != nil {
-		return err
-	}
+	if option.Config.EnableIPv4 {
+		matchFromIPSecEncrypt := fmt.Sprintf("%#08x/%#08x", linux_defaults.RouteMarkDecrypt, linux_defaults.RouteMarkMask)
+		matchFromIPSecDecrypt := fmt.Sprintf("%#08x/%#08x", linux_defaults.RouteMarkEncrypt, linux_defaults.RouteMarkMask)
 
-	if option.Config.Masquerade {
-		ingressSnatSrcAddrExclusion := node.GetHostMasqueradeIPv4().String()
-		if option.Config.Tunnel == option.TunnelDisabled {
-			ingressSnatSrcAddrExclusion = node.GetIPv4ClusterRange().String()
-		}
-
-		// Masquerade all traffic from the host into the ifName
-		// interface if the source is not the internal IP
-		//
-		// The following conditions must be met:
-		// * Must be targeted for the ifName interface
-		// * Must be targeted to an IP that is not local
-		// * Tunnel mode:
-		//   * May not already be originating from the masquerade IP
-		// * Non-tunnel mode:
-		//   * May not orignate from any IP inside of the cluster range
+		// Clear the Kubernetes masquerading mark bit to skip source PAT
+		// performed by kube-proxy for all packets destined for Cilium. Cilium
+		// installs a dedicated rule which does the source PAT to the right
+		// source IP.
+		clearMasqBit := fmt.Sprintf("%#08x/%#08x", 0, proxy.MagicMarkK8sMasq)
 		if err := runProg("iptables", []string{
-			"-t", "nat",
-			"-A", ciliumPostNatChain,
-			"!", "-s", ingressSnatSrcAddrExclusion,
-			"!", "-d", node.GetIPv4AllocRange().String(),
+			"-t", "mangle",
+			"-A", ciliumPostMangleChain,
+			"-o", ifName,
 			"-m", "mark", "!", "--mark", matchFromIPSecDecrypt, // Don't match ipsec traffic
 			"-m", "mark", "!", "--mark", matchFromIPSecEncrypt, // Don't match ipsec traffic
-			"-o", ifName,
-			"-m", "comment", "--comment", "cilium host->cluster masquerade",
-			"-j", "SNAT", "--to-source", node.GetHostMasqueradeIPv4().String()}, false); err != nil {
+			"-m", "comment", "--comment", "cilium: clear masq bit for pkts to " + ifName,
+			"-j", "MARK", "--set-xmark", clearMasqBit}, false); err != nil {
 			return err
 		}
 
-		// Masquerade all traffic from the host into the ifName
-		// interface if the source is 127.0.0.1
+		// kube-proxy does not change the default policy of the FORWARD chain
+		// which means that while packets to services are properly DNAT'ed,
+		// they are later dropped in the FORWARD chain. The issue has been
+		// resolved in #52569 and will be fixed in k8s >= 1.8. The following is
+		// a workaround for earlier Kubernetes versions.
 		//
```

Comment:
Just curious if anyone know if k8s 1.8+ has the fix or not?

Reply:
@aanm Do you know?

Reply:
the commit is available in >=v1.9 https://github.com/kubernetes/kubernetes/commit/2f622b2a2858eea3b74a9f83a368197719b06a6f

@tgraf maybe we can remove this since we are dropping support for 1.8? Does it hurt having this just in case there's a regression in k8s?

Reply:
I'm fine to remove it. The fewer rules the better.